evaluation/eval.py

```diff
@@ -37,65 +32,63 @@ class EvaluationArguments:
         default=None,
         metadata={"help": "Pretrained tokenizer name or path if not the same as model_name."}
     )
-    output_dir: Optional[str] = field(
+    output_dir: str = field(
         default="outputs",
         metadata={"help": "Directory for saving evaluation outputs."}
     )
-    random_seed: Optional[int] = field(
+    tag: Optional[str] = field(
+        default=None,
+        metadata={"help": "Identifier for the evaluation run."}
+    )
+    random_seed: int = field(
         default=24,
         metadata={"help": "Customized random seed"}
     )
     eval_tasks: Optional[List[str]] = field(
         default=None,
         metadata={"help": "A list of tasks to run the evaluation on, e.g. tydiqa_secondary"}
     )
-
-
-def main():
-    # parse arguments
-    parser = HfArgumentParser(EvaluationArguments)
-    eval_args, = parser.parse_args_into_dataclasses()
-
-    # set up logging
-    logging.basicConfig(
-        format="%(asctime)s - %(levelname)s - %(name)s -   %(message)s",
-        datefmt="%m/%d/%Y %H:%M:%S",
+    device: str = field(
+        default="cuda",
+        metadata={"help": "Device on which to run evaluation"}
```

Comment:
`device` comes with the TrainingArguments class in HF. We might want to just use that instead.

Reply:
So do you suggest that we do 

```python
# assume imports
parser = HfArgumentParser((EvaluationArguments, TrainingArguments))
eval_args, train_args = parser.parse_args_into_dataclasses()
```

We could do that, but part of me also thinks doing this just for the sake of `device` might a bit too much? Unless we forsee other flags in `TrainingArguments` being used in the future. What do you think?

Reply:
Yeah, that's what I had in mind. I absolutely agree that it feels overkill. But then I looked at the implementation of TrainingArguments (see `def _setup_devices(self) -> "torch.device"` in [here](https://huggingface.co/transformers/_modules/transformers/training_args.html)) and thought it might just be worth it. 

I'm honestly fine with either. What do you think?

Reply:
Yes, the `device` specification does look thorough. Also, I see some useful arguments we could also probably use for evaluation, particularly ones relating to the data loader. I'll push up a commit that adds `TrainingArguments` as per your suggestion!